src/_P036_FrameOLED.ino

```diff
@@ -791,6 +761,109 @@ boolean Plugin_036(uint8_t function, struct EventStruct *event, String& string)
+{
+  if (UserVar[event->BaseVarIndex] == 1) {
+    // Display is on.
+    ScrollingPages.Scrolling = 1; // page scrolling running -> no line scrolling allowed
+    NFrames = P36_Nlines / ScrollingPages.linesPerFrame;
+    OLEDIndex = PCONFIG(7);
+    HeaderContent = static_cast<eHeaderContent>(get8BitFromUL(PCONFIG_LONG(0), 8));             // Bit15-8 HeaderContent
+    HeaderContentAlternative = static_cast<eHeaderContent>(get8BitFromUL(PCONFIG_LONG(0), 0));  // Bit 7-0 HeaderContentAlternative
+
+    //      Now create the string for the outgoing and incoming frames
+    String tmpString;
+    tmpString.reserve(P36_NcharsV1);
+
+    //      Construct the outgoing string
+    for (uint8_t i = 0; i < ScrollingPages.linesPerFrame; i++)
+    {
+      tmpString = String(P036_DisplayLinesV1[(ScrollingPages.linesPerFrame * frameCounter) + i].Content);
+      ScrollingPages.oldString[i] = P36_parseTemplate(tmpString, 20);
+    }
+
+    // now loop round looking for the next frame with some content
+    //   skip this frame if all lines in frame are blank
+    // - we exit the while loop if any line is not empty
+    boolean foundText = false;
+    int ntries = 0;
+    while (!foundText) {
+
+      //        Stop after framecount loops if no data found
+      ntries += 1;
+      if (ntries > NFrames) break;
+
+      if (nextFrameToDisplay == 0xff) {
+        // Increment the frame counter
+        frameCounter = frameCounter + 1;
+        if ( frameCounter > NFrames - 1) {
+          frameCounter = 0;
+          currentFrameToDisplay = 0;
+        }
+      }
+      else {
+        // next frame because content changed in PLUGIN_WRITE
+        frameCounter = nextFrameToDisplay;
+      }
+
+      //        Contruct incoming strings
+      for (uint8_t i = 0; i < ScrollingPages.linesPerFrame; i++)
+      {
+        tmpString = String(P036_DisplayLinesV1[(ScrollingPages.linesPerFrame * frameCounter) + i].Content);
+        ScrollingPages.newString[i] = P36_parseTemplate(tmpString, 20);
+        if (ScrollingPages.newString[i].length() > 0) foundText = true;
+      }
+      if (foundText) {
+        if (nextFrameToDisplay == 0xff) {
+          if (frameCounter != 0) {
+            ++currentFrameToDisplay;
+          }
+        }
+        else currentFrameToDisplay = nextFrameToDisplay;
+      }
+    }
+    nextFrameToDisplay = 0xFF;
+
+    if ((currentFrameToDisplay + 1) > nrFramesToDisplay) {
+      nrFramesToDisplay = currentFrameToDisplay + 1;
+    }
+
+    // Update max page count
+    if (MaxFramesToDisplay == 0xFF) {
+      // not updated yet
+      for (uint8_t i = 0; i < NFrames; i++) {
+        for (uint8_t k = 0; k < ScrollingPages.linesPerFrame; k++)
+        {
+          tmpString = String(P036_DisplayLinesV1[(ScrollingPages.linesPerFrame * i) + k].Content);
+          tmpString = P36_parseTemplate(tmpString, 20);
+          if (tmpString.length() > 0) {
+            // page not empty
+            MaxFramesToDisplay ++;
```

Comment:
`MaxFramesToDisplay` is set to `0xFF` here, so please set it first to 0 before starting to ++ the value.
If someone changes the type to something other than `uint8_t`, this will yield very unexpected behavior.

Reply:
This is existing code, I only moved it to a function to be able to call it from multiple places.
Sounds indeed plausible, so I'll improve that.